drivers/gpio/gpio_xlnx_ps.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Xilinx Processor System MIO / EMIO GPIO controller driver
+ * Parent (IRQ handler) module
+ *
+ * Copyright (c) 2022, Weidmueller Interface GmbH & Co. KG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+
+#include <drivers/gpio.h>
+#include "gpio_utils.h"
+#include "gpio_xlnx_ps.h"
+#include "gpio_xlnx_ps_bank.h"
+
+#define LOG_MODULE_NAME gpio_xlnx_ps
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define DT_DRV_COMPAT xlnx_ps_gpio
+
+/*
+ * An API is required for this driver, but as no pin access is provided at
+ * this level, use the default API contents provided by the driver subsystem.
+ */
+static const struct gpio_driver_api gpio_xlnx_ps_default_apis;
+
+/**
+ * @brief Initialize a Xilinx PS GPIO controller parent device
+ *
+ * Initialize a Xilinx PS GPIO controller parent device, whose task it is
+ * to handle the IRQ line of each controller instance, while the configuration,
+ * status and data acquisition of each MIO / EMIO GPIO pin associated with
+ * the parent controller instance is handled via the respective GPIO pin
+ * bank's child device.
+ *
+ * @param dev Pointer to the PS GPIO controller's device.
+ *
+ * @retval Always 0.
+ */
+static int gpio_xlnx_ps_init(const struct device *dev)
+{
+	const struct gpio_xlnx_ps_dev_cfg *dev_conf = dev->config;
+
+	/* Initialize the device's interrupt */
+	dev_conf->config_func(dev);
+
+	return 0;
+}
+
+/**
+ * @brief Xilinx PS GPIO controller parent device ISR
+ *
+ * Interrupt service routine for the Xilinx PS GPIO controller's
+ * IRQ. The ISR iterates all associated MIO / EMIO GPIO pink bank
+ * child device instances and checks each bank's interrupt status.
+ * If any pending interrupt is detected within a GPIO pin bank,
+ * the callbacks registered for the respective bank are triggered
+ * using the functionality provided by the GPIO sub-system.
+ *
+ * @param dev Pointer to the PS GPIO controller's device.
+ */
+static void gpio_xlnx_ps_isr(const struct device *dev)
+{
+	const struct gpio_xlnx_ps_dev_cfg *dev_conf = dev->config;
+
+	struct gpio_driver_api *api;
+	struct gpio_xlnx_ps_bank_dev_data *bank_data;
+
+	uint32_t bank;
+	uint32_t int_mask;
+
+	for (bank = 0; bank < dev_conf->num_banks; bank++) {
+		api = (struct gpio_driver_api *)dev_conf->bank_devices[bank]->api;
```

Comment:
```suggestion
		api = dev_conf->bank_devices[bank]->api;
```

cast no longer necessary when `api` is declared const.

Reply:
removed type cast as suggested.